File: src/Entity/Commande.php
<?php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=CommandeRepository::class)
 */
class Commande
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     * @Assert\NotBlank(message="Le champs adressse de livraison est obligatoire * ")
     */
    private $adresseLivraison;

    /**
     * @ORM\Column(type="float")
     * @Assert\Regex("#[0-9]+(\.[0-9]{1,2})?#")
     */
    private $totalCommande;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $modeLivraison;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $renseignement;

    /**
     * @ORM\Column(type="boolean", nullable=true)
     */
    private $status=1;

    /**
     * @ORM\ManyToOne(targetEntity=Coupon::class, cascade={"persist", "remove"},inversedBy="commande")
     */
    private $code;

    /**
     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="commandes")
     */
    private $User;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getAdresseLivraison(): ?string
    {
        return $this->adresseLivraison;
    }

    public function setAdresseLivraison(?string $adresseLivraison): self
    {
        $this->adresseLivraison = $adresseLivraison;

        return $this;
    }

    public function getTotalCommande(): ?float
    {
        return $this->totalCommande;
    }

    public function setTotalCommande(float $totalCommande): self
    {
        $this->totalCommande = $totalCommande;

        return $this;
    }

    public function getModeLivraison(): ?string
    {
        return $this->modeLivraison;
    }

    public function setModeLivraison(string $modeLivraison): self
    {
        $this->modeLivraison = $modeLivraison;

        return $this;
    }

    public function getRenseignement(): ?string
    {
        return $this->renseignement;
    }

    public function setRenseignement(?string $renseignement): self
    {
        $this->renseignement = $renseignement;

        return $this;
    }

    public function getStatus(): ?bool
    {
        return $this->status;
    }

    public function setStatus(bool $status): self
    {
        $this->status = $status;

        return $this;
    }

    public function getCode()
    {
        return $this->code;
    }
    public function setCode($code)
    {
        return $this->code=$code;
    }

    function addCodeCoupon(Coupon $code)
    {
        return $this->code->add($code);

    }

    public function getUser(): ?User
    {
        return $this->User;
    }

    public function setUser(?User $User): self
    {
        $this->User = $User;

        return $this;
    }
}

File: src/com/mycompagny/entities/Commande.java
package com.mycompagny.entities;


public class Commande {

    private int id;
    private String adresseLivraison;
    private Float totalCommande;
    private String modeLivraison;
    private String renseignement;
    private boolean status;
    private int iduser;
    private String user;

    public Commande() {
    }

    public Commande(String adresseLivraison, Float totalCommande, String modeLivraison, String renseignement, boolean status) {
        this.adresseLivraison = adresseLivraison;
        this.totalCommande = totalCommande;
        this.modeLivraison = modeLivraison;
        this.renseignement = renseignement;
        this.status = status;
    }

    public Commande(int id, String user,String adresseLivraison, Float totalCommande, String modeLivraison, String renseignement, boolean status) {
        this.id = id;
        this.user = user;
        this.adresseLivraison = adresseLivraison;
        this.totalCommande = totalCommande;
        this.modeLivraison = modeLivraison;
        this.renseignement = renseignement;
        this.status = status;
    }

    public Commande(String adresseLivraison, Float totalCommande, String modeLivraison, String renseignement, boolean status, int iduser) {
        this.adresseLivraison = adresseLivraison;
        this.totalCommande = totalCommande;
        this.modeLivraison = modeLivraison;
        this.renseignement = renseignement;
        this.status = status;
        this.iduser = iduser;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getAdresseLivraison() {
        return adresseLivraison;
    }

    public void setAdresseLivraison(String adresseLivraison) {
        this.adresseLivraison = adresseLivraison;
    }

    public Float getTotalCommande() {
        return totalCommande;
    }

    public void setTotalCommande(Float totalCommande) {
        this.totalCommande = totalCommande;
    }

    public String getModeLivraison() {
        return modeLivraison;
    }

    public void setModeLivraison(String modeLivraison) {
        this.modeLivraison = modeLivraison;
    }

    public String getRenseignement() {
        return renseignement;
    }

    public void setRenseignement(String renseignement) {
        this.renseignement = renseignement;
    }

    public boolean getStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public int getIduser() {return iduser;}

    public void setIduser(int iduser) {this.iduser = iduser;}

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }
}
